packages/back-end/src/models/ApiKeyModel.ts

```diff
@@ -28,7 +29,13 @@ const apiKeySchema = new mongoose.Schema({
 
 type ApiKeyDocument = mongoose.Document & ApiKeyInterface;
 
-const ApiKeyModel = mongoose.model<ApiKeyDocument>("ApiKey", apiKeySchema);
+const ApiKeyModel = mongoose.model<ApiKeyInterface>("ApiKey", apiKeySchema);
```

Comment:
The [Mongoose TypeScript docs](https://mongoosejs.com/docs/typescript.html) explicitly call out that we should be using our interface instead of a union type with the mongoose `Document` as the generic argument to the model, so I've made that change in all files.